google/cloud/bigquery/job/base.py

```diff
@@ -74,8 +75,20 @@ def _error_result_to_exception(error_result):
     status_code = _ERROR_REASON_TO_EXCEPTION.get(
         reason, http.client.INTERNAL_SERVER_ERROR
     )
+    # Manually create error message to preserve both error_result and errors.
+    # Can be removed once b/310544564 and b/318889899 are resolved.
+    string = ""
+    if errors:
+        string = "; "
+        for err in errors:
+            for key, value in err.items():
+                string += key + ": " + str(value) + ", "
+        string = string[:-2]
```

Comment:
I feel like a comment here quickly makes it obvious why we are using a slice here.

```suggestion
        string = string[:-2]  # strips off the last unneeded comma and space
```